packages/Meshfree/test/PointCloudProblemGenerator/ExodusProblemGenerator.hpp

```diff
@@ -81,24 +81,32 @@ class ExodusProblemGenerator
                             const std::string &target_exodus_file );
 
     // Create a problem where all points are uniquely owned (i.e. no
-    // ghosting). Both source and target fields are initialized to zero.
+    // ghosting). Both source and target fields have one component and are
+    // initialized to zero.
     void createUniquelyOwnedProblem(
         Kokkos::View<Coordinate **, Kokkos::LayoutLeft, SourceDevice>
             &src_coords,
+        Kokkos::View<Scalar **, Kokkos::LayoutLeft, SourceDevice> &src_field,
         Kokkos::View<Coordinate **, Kokkos::LayoutLeft, TargetDevice>
-            &tgt_coords ) override;
+            &tgt_coords,
+        Kokkos::View<Scalar **, Kokkos::LayoutLeft, TargetDevice> &tgt_field )
+        override;
 
     // Create a general problem where points may exist on multiple
-    // processors. Both source and target fields are initialized to zero.
+    // processors. Both source and target fields have 1 component and are
```

Comment:
Why just one?  I thought exodus files could include solution files.

Reply:
I wasn't aware of solution files.

In this case the basic exodus generator just reads a mesh and allocates a field of a single component as it has no internal solution state. Other implementations of the interface could return fields of multiple components. If we generate some exodus solution files in the future we could refactor this to return fields with dimensions that we get from the solutions.